core/src/main/java/org/elasticsearch/index/search/stats/SearchSlowLog.java

```diff
@@ -68,60 +64,56 @@
-            }
+        long fetchWarnThreshold = settings.getAsTime(INDEX_SEARCH_SLOWLOG_THRESHOLD_FETCH_WARN, TimeValue.timeValueNanos(SearchSlowLog.this.fetchWarnThreshold)).nanos();
+        if (fetchWarnThreshold != SearchSlowLog.this.fetchWarnThreshold) {
+            SearchSlowLog.this.fetchWarnThreshold = fetchWarnThreshold;
+        }
+        long fetchInfoThreshold = settings.getAsTime(INDEX_SEARCH_SLOWLOG_THRESHOLD_FETCH_INFO, TimeValue.timeValueNanos(SearchSlowLog.this.fetchInfoThreshold)).nanos();
+        if (fetchInfoThreshold != SearchSlowLog.this.fetchInfoThreshold) {
+            SearchSlowLog.this.fetchInfoThreshold = fetchInfoThreshold;
+        }
+        long fetchDebugThreshold = settings.getAsTime(INDEX_SEARCH_SLOWLOG_THRESHOLD_FETCH_DEBUG, TimeValue.timeValueNanos(SearchSlowLog.this.fetchDebugThreshold)).nanos();
+        if (fetchDebugThreshold != SearchSlowLog.this.fetchDebugThreshold) {
+            SearchSlowLog.this.fetchDebugThreshold = fetchDebugThreshold;
+        }
+        long fetchTraceThreshold = settings.getAsTime(INDEX_SEARCH_SLOWLOG_THRESHOLD_FETCH_TRACE, TimeValue.timeValueNanos(SearchSlowLog.this.fetchTraceThreshold)).nanos();
+        if (fetchTraceThreshold != SearchSlowLog.this.fetchTraceThreshold) {
+            SearchSlowLog.this.fetchTraceThreshold = fetchTraceThreshold;
+        }
 
-            String level = settings.get(INDEX_SEARCH_SLOWLOG_LEVEL, ShardSlowLogSearchService.this.level);
-            if (!level.equals(ShardSlowLogSearchService.this.level)) {
-                ShardSlowLogSearchService.this.queryLogger.setLevel(level.toUpperCase(Locale.ROOT));
-                ShardSlowLogSearchService.this.fetchLogger.setLevel(level.toUpperCase(Locale.ROOT));
-                ShardSlowLogSearchService.this.level = level;
-            }
+        String level = settings.get(INDEX_SEARCH_SLOWLOG_LEVEL, SearchSlowLog.this.level);
+        if (!level.equals(SearchSlowLog.this.level)) {
+            SearchSlowLog.this.queryLogger.setLevel(level.toUpperCase(Locale.ROOT));
+            SearchSlowLog.this.fetchLogger.setLevel(level.toUpperCase(Locale.ROOT));
+            SearchSlowLog.this.level = level;
+        }
 
-            boolean reformat = settings.getAsBoolean(INDEX_SEARCH_SLOWLOG_REFORMAT, ShardSlowLogSearchService.this.reformat);
-            if (reformat != ShardSlowLogSearchService.this.reformat) {
-                ShardSlowLogSearchService.this.reformat = reformat;
-            }
+        boolean reformat = settings.getAsBoolean(INDEX_SEARCH_SLOWLOG_REFORMAT, SearchSlowLog.this.reformat);
+        if (reformat != SearchSlowLog.this.reformat) {
+            SearchSlowLog.this.reformat = reformat;
         }
     }
 
-    @Inject
-    public ShardSlowLogSearchService(ShardId shardId, @IndexSettings Settings indexSettings, IndexSettingsService indexSettingsService) {
-        super(shardId, indexSettings);
+    SearchSlowLog(ESLogger logger, Settings indexSettings) {
```

Comment:
same issue caused by the package change. I think we should keep the log name as `index.search.slowlog` . To do so it will require to move the SearchSlowLog to the index.search package, together with ShardSearchService . I think that's OK?